base/sparse/cholmod.jl

```diff
@@ -860,7 +860,18 @@ convert(::Type{Dense}, A::Sparse) = sparse_to_dense(A)
 function (::Type{Sparse}){Tv<:VTypes}(m::Integer, n::Integer,
         colptr::Vector{SuiteSparse_long}, rowval::Vector{SuiteSparse_long},
         nzval::Vector{Tv}, stype)
-    # check if columns are sorted
+    # checks
+    ## length of input
+    if length(colptr) <= n
+        throw(ArgumentError("length of colptr must be at least n + 1 = $(n + 1) but was $(length(colptr))"))
+    end
+    if colptr[n + 1] > length(rowval)
```

Comment:
> This constructor is zero based. I'm wondering if it should have a more alarming name for that reason.

That sounds great! Similarly, if some instances of `colptr` are zero based, might be worth similarly naming them appropriately? Best!